### PR TITLE
Styleci bridge

### PR DESCRIPTION
**Pull request by Soullivaneuh**
I saw that you are using both StyleCI and PHP-CS-Fixer.

I propose you to use this bridge that will parse your `.styleci.yml` file and configure your cs-fixer accordingly.

The goal of this PR is to have only one configuration file to maintain.

Note: If you already configured StyleCI on web interface, I suggest you to delete it except for the header_comment section.

### CODE REVIEW

File: src/Helper/GitHelper.php [hunk 1]
```diff
@@ -420,7 +420,7 @@ public function mergeBranchWithLog($base, $sourceBranch, $commitMessage, $source
         }
 
         $this->processHelper->runCommand(
-            ['git', 'merge', '--no-ff', '--log' , '--no-commit', $sourceBranch]
+            ['git', 'merge', '--no-ff', '--log', '--no-commit', $sourceBranch]
```
Comment (cordoval) [inline src/Helper/GitHelper.php]: good catch @Soullivaneuh

**Comment by cordoval** [comment]
cc @GrahamCampbell what do you think of this? is @Soullivaneuh your coworker? I am :+1: for this if it makes sense to you, honestly i would rather go simple, but if this is it then fine.

**Comment by GrahamCampbell** [comment]
No, @Soullivaneuh isn't my co-worker. ;)

I think this is probably a good idea, at least for now, but the plans for StyleCI in the future will extend behond php-cs-fixer.

**Comment by GrahamCampbell** [comment]
NB, due to time constraints, StyleCI is unlikely to go through any major changes until July 2016, so :+1: for now. :)

**Comment by Soullivaneuh** [comment]
> is @Soullivaneuh your coworker?

Ha ha no, but he ping me some time on my project. :-)

> but the plans for StyleCI in the future will extend behond php-cs-fixer.

I'm curious to see that. :-) Still compatible with it I hope! ;-)

**Comment by cordoval** [comment]
I am again divided on this PR. My vote is :-1: in its present state of this PR. If i do this it would be an external tool downloaded via bldr. I cannot plug a dependency just for this. I was even very reluctant of styleCI, that is why it is configured only via web and it is not intruding the repository. The right way probably will be add a bldr task to download and/or run the cli tool and make it download and plug it via php-cs-fixer somehow if that is possible. Bldr does not taint my composer.json for instance but yeah i know it adds a bunch of files already to the root project folder. So no cannot do this, sorry, thank you though.

**Comment by cordoval** [comment]
cc @sstok feel free to reopen if you think is debatable

File: src/Helper/GitHelper.php (hunk 1 above)
Comment (cordoval) [inline src/Helper/GitHelper.php]: addressed on 2f51d20
thanks @Soullivaneuh

**Comment by Soullivaneuh** [comment]
> I cannot plug a dependency just for this.

Sorry I don't understand. It's just a composer dependencies that will works automatically once installed, nothing else to do. BTW, this is a very small size project and it's required only for dev. Users wont be forces to download this library.

>  I was even very reluctant of styleCI

Just for curiosity, why?

> that is why it is configured only via web and it is not intruding the repository

Well, IMHO it's easier to manage a config file directly trough the repository. Because contributors can't see and apply your configuration actually.

In my personal case, and I'm not the only one, I largely prefer run a fixer cmd line instead of download a patch or apply it via GitHub. That's why I wrote this bridge.

**Comment by sstok** [comment]
> Sorry I don't understand. It's just a composer dependencies that will works automatically once installed, nothing else to do. BTW, this is a very small size project and it's required only for dev. Users wont be forces to download this library.

Even though it's a small the dependencies are not.
https://github.com/Soullivaneuh/php-cs-fixer-styleci-bridge/blob/master/composer.json#L16-L19

The PHP-cs-fixer can be installed as phar or Composer global and you can reuse it for any project.
This library is installed for every project you use it in, even though it's only used for cs and nothing else.
So it's pretty heavy for what you use it.

Gush is a single repository so the amount of maintenance for keeping the configuration in-sync is no problem for us. If you maintain a big number of (sub)projects I can imagine this can be very useful :+1: 

> I was even very reluctant of styleCI
>> Just for curiosity, why?

In Holland we have saying for about this: "onbekend maakt onbemind."
If you don't know it, you won't like it.

StyleCI is still very new to some, and granting access to a service you don't know nothing about is a security risk. Note that StyleCI has Push access (for pull-requests).


On a side-note, is it possible to configure the configure the header in a config file (if not can be this added, or are there to many security issues)? /cc @GrahamCampbell 

Now I need to go the web configuration, but [generating](https://github.com/rollerworks/SkeletonDancer) config files is so much easier :)

**Comment by Soullivaneuh** [comment]
> Even though it's a small the dependencies are not.
https://github.com/Soullivaneuh/php-cs-fixer-styleci-bridge/blob/master/composer.json#L16-L19

I don't see the problem for those small symfony deps. It will be kept on lowest at minimum will it's maintained, and any longer as is possible.

> In Holland we have saying for about this: "onbekend maakt onbemind."
If you don't know it, you won't like it.

Yeah, but you're using it actually, so.. what? :-)

> On a side-note, is it possible to configure the configure the header in a config file (if not can be this added, or are there to many security issues)? /cc @GrahamCampbell


I know the answer for the moment: No. See: https://github.com/Soullivaneuh/php-cs-fixer-styleci-bridge#header-comment

But I agree with you, it should be added.

**Comment by sstok** [comment]
I am, but cordoval is the owner of the GushPHP organization ;)

**Pull request closed without merging**